trackers/javascript-tracker/test/unit/in_queue.test.ts

```diff
@@ -183,4 +211,58 @@ describe('InQueueManager', () => {
       ]);
     }).not.toThrow();
   });
+
+  it('A custom callback with arguments provided will pass those arguments into the callback parameters', () => {
+    asyncQueue.push([
+      function (a: number, b: number) {
+        expect(a).toEqual(1);
+        expect(b).toEqual(2);
+      },
+      1,
+      2,
+    ]);
+  });
+
+  it('The callback will be passed the tracker dictionary as the final argument', () => {
+    asyncQueue.push([
+      function (a: number, b: number, trackers: Record<string, any>) {
+        expect(a).toEqual(1);
+        expect(b).toEqual(2);
+        expect(trackers.sp1).toEqual(mockTrackers.callback_sp1);
+      },
+      1,
+      2,
+    ]);
+  });
+
+  it('The callback will be passed the tracker dictionary as the argument if there is only one parameter', () => {
+    asyncQueue.push([
+      function (trackers: Record<string, any>) {
+        const tracker = trackers.sp1;
+        expect(tracker).toEqual(mockTrackers.callback_sp1);
+      },
+    ]);
+  });
+
+  it('The callback can access the tracker dictionary using both `this` and the first argument', () => {
+    asyncQueue.push([
+      function (this: any, trackers: Record<string, any>) {
+        expect(this.sp1).toEqual(mockTrackers.callback_sp1);
+        expect(trackers.sp1).toEqual(mockTrackers.callback_sp1);
+      },
+    ]);
+  });
+
+  it('The callback can access the tracker dictionary using both `this` and  the last argument, along with arguments', () => {
+    asyncQueue.push([
+      function (this: any, a: number, b: number, trackers: Record<string, any>) {
+        expect(this.sp1).toEqual(mockTrackers.callback_sp1);
+        expect(a).toEqual(1);
+        expect(b).toEqual(2);
+        expect(trackers.sp1).toEqual(mockTrackers.callback_sp1);
+      },
+      1,
+      2,
+    ]);
+  });
```

Comment:
I think you could get by with just 1/3/5 of these; 2 & 4 seem redundant. (1 & 3 are also replicated by 5 but show different use cases that are kind valuable to have as exposition).